examples/wifi-echo/server/esp32/Makefile

```diff
@@ -26,9 +26,9 @@ EXTRA_COMPONENT_DIRS += $(PROJECT_PATH)/third_party/connectedhomeip/config/esp32
                         $(PROJECT_PATH)/../../../common/QRCode \
                         $(PROJECT_PATH)/../../../common/screen-framework \
 
-CXXFLAGS += -std=c++11 -Os
-CPPFLAGS += -Os
-CFLAGS += -Os
+CXXFLAGS += -std=c++11 -Os -DLWIP_IPV6_SCOPES=0
```

Comment:
Is there a way to perform this configuration via the sdkconfig files? Seems very _not_ esp-idf like to do it this way.

Reply:
@sagar-apple IMO, the option LWIP_IPV6_SCOPES should not be disabled, neither through CFLAGS nor sdkconfig. The reasons being:
1. LWIP mentions that this option is to be disabled only for single-interface configurations, see [here](https://github.com/espressif/esp-lwip/blob/2.1.2-esp/src/include/lwip/opt.h#L2430)
2. ESP-IDF esp-netif layer, the newly added one in v4.1, assumes that zone (introduced when LWIP_IPV6_SCOPES is enabled) is present in the IPv6 address

With the current delta in LWIP for ESP-IDF and CHIP, I think we can modify the test memcmp statement to just compare member `addr` of ip6_addr_t instead. Thoughts?

Reply:
Yeah I guess we could update the test instead. Since @rwalker-apple has looked at this test already maybe there's a reason he opted to not do it that way?

Reply:
Modifying the memcmp() is insufficient because CHIP thinks ip6_addr_t is 16 bytes, allocates 16 bytes for it (as Addr[] in IPAddress) and routinely moves the memory as *(ip6_addr_t *)&Addr, which moves _**20 bytes**_.  This can easily lead to memory corruption.

My change is a hack, needs to be addressed with a more complete fix if we are planning to use LWIP_IPV6_SCOPES, which looks to me like a breaking change on LWIP's part...